stablehlo/transforms/ShapeLegalizeToStablehlo.cpp

```diff
@@ -118,6 +118,16 @@ Value castToIndex(PatternRewriter& rewriter, Location loc, Value value) {
   return cast.getResult(0);
 }
 
+void insertShapeAssertionCustomCall(OpBuilder builder, Location loc,
+                                    Value assert) {
+  auto customCall = builder.create<stablehlo::CustomCallOp>(loc, TypeRange{},
+                                                            ValueRange{assert});
+  customCall.setCallTargetName("shape_assertion");
+  customCall.setHasSideEffect(true);
+  customCall->setAttr("error_message",
+                      builder.getStringAttr("Shape assertion failed"));
```

Comment:
Who/what consumes this? How do we know that this is the format expected by downstream consumers?

Reply:
This particular signature of the custom_call is used for TF shape refinement pass.
[Here](https://github.com/openxla/xla/blob/91713eed21f5cfd58c64a68c13bb89cc40b67229/xla/python/refine_polymorphic_shapes.cc#L58) we can find the reliance on this particular custom_call

and 

[Here]() we can see that in action. 
(RefinePolymorphicShapes)[https://github.com/openxla/xla/blob/91713eed21f5cfd58c64a68c13bb89cc40b67229/xla/python/refine_polymorphic_shapes.cc#L244)   is used to refine dynamic shape in StableHLO embedded in [XlaCallModule]
(https://github.com/tensorflow/tensorflow/blob/a632c89dd7789b62aa69ada8f1bb689fc44cbb8d/tensorflow/compiler/tf2xla/kernels/xla_call_module_loader.cc#L409). 

The main interface being [InferModuleShape](https://github.com/tensorflow/tensorflow/blob/a632c89dd7789b62aa69ada8f1bb689fc44cbb8d/tensorflow/compiler/mlir/tensorflow/transforms/shape_inference.h#L49)

Reply:
Cool, makes sense, thanks! Feels brittle, but I don't necessarily have a better suggestion.